hummingbot/connector/exchange_base_v2.py

```diff
@@ -0,0 +1,815 @@
+
+    async def check_network(self) -> NetworkStatus:
+        """
+        Checks connectivity with the exchange using the API
+        """
+        try:
+            await self._api_get(path_url=self.CHECK_NETWORK_URL)
+        except asyncio.CancelledError:
+            raise
+        except Exception:
+            return NetworkStatus.NOT_CONNECTED
+        return NetworkStatus.CONNECTED
+
+    # loops and sync related methods
+    #
+    async def _trading_rules_polling_loop(self):
+        """
+        Updates the trading rules by requesting the latest definitions from the exchange.
+        Executes regularly every 30 minutes
+        """
+        while True:
+            try:
+                await safe_gather(self._update_trading_rules())
+                await asyncio.sleep(self.TRADING_RULES_INTERVAL)
+            except asyncio.CancelledError:
+                raise
+            except Exception:
+                self.logger().network(
+                    "Unexpected error while fetching trading rules.", exc_info=True,
+                    app_warning_msg=f"Could not fetch new trading rules from {self.name_cap}"
+                                    " Check network connection.")
+                await asyncio.sleep(0.5)
+
+    async def _trading_fees_polling_loop(self):
+        """
+        Only some exchanges provide a fee endpoint.
+        If _update_trading_fees() is not defined, we just exit the loop
+        """
+        while True:
+            try:
+                await safe_gather(self._update_trading_fees())
+                await self._sleep(self.TRADING_FEES_INTERVAL)
+            except NotImplementedError:
+                return
+            except asyncio.CancelledError:
+                raise
+            except Exception:
+                self.logger().network(
+                    "Unexpected error while fetching trading fees.", exc_info=True,
+                    app_warning_msg=f"Could not fetch new trading fees from {self.name_cap}."
+                                    " Check network connection.")
+                await self._sleep(0.5)
+
+    async def _status_polling_loop(self):
+        """
+        Performs all required operation to keep the connector updated and synchronized with the exchange.
+        It contains the backup logic to update status using API requests in case the main update source
+        (the user stream data source websocket) fails.
+        It also updates the time synchronizer. This is necessary because the exchange requires
+        the time of the client to be the same as the time in the exchange.
+        Executes when the _poll_notifier event is enabled by the `tick` function.
+        """
+        while True:
+            try:
+                await self._poll_notifier.wait()
+                await self._update_time_synchronizer()
+
+                # the following method is implementation-specific
+                await self._status_polling_loop_fetch_updates()
```

Comment:
`_status_polling_loop_fetch_updates` should be implemented in this class as an abstract method